Assets/Scripts/Core/ShipModel/ShipParameters.cs

```diff
@@ -60,24 +65,281 @@ public class ShipParameters {
         public float totalBoostRotationalTime;
         public float totalBoostTime;
         public float yawMultiplier;
+        public float boostDivertEfficiency;
+
+        
 
         public string ToJsonString() {
-            return JsonConvert.SerializeObject(this, Formatting.Indented);
+                return JsonConvert.SerializeObject(this, Formatting.Indented);
         }
 
         [CanBeNull]
         public static ShipParameters FromJsonString(string json) {
             try {
-                var parameters = JsonConvert.DeserializeObject<ShipParameters>(json);
-                if (parameters?.maxAngularVelocity == 0) parameters.maxAngularVelocity = Defaults.maxAngularVelocity;
-                if (parameters?.boostSpoolUpTime == 0) parameters.boostSpoolUpTime = Defaults.boostSpoolUpTime;
+                ShipParameters returnedParameters = ShipParameters.Defaults;
+
+                var parameters = JsonConvert.DeserializeObject<NullableParameters>(json);
+
+                if (parameters.angularDrag != Defaults.angularDrag)
```

Comment:
there looked to be something much cleaner and simpler here before, what happened? 

this looks like a lot of duplicated logic! I wonder if that can be combined into a function... (I have no idea how to reference object variable names as parameters but I'm almost certain there'll be a way).

Alternatively, I think you can mark json fields in the object with [Default] attributes or something to do similar logic to this (IIRC this is why exported level json omits zero values on coordinates, the default attribute just ignores them)

Reply:
I apologize for this mess, the original code turned out to be completely broken, where the only reason it didnt crash was because it didn't run at all.  I probably should not have submitted this mess, but I mostly work on this when it is late, and I wanted it fixed and sleep. 

I am working on something much cleaner now, but that also is giving me some problems for now. Il try to get that submitted.